lib/utils.js

```diff
@@ -239,3 +245,79 @@ function routeResolvedFile (router, path, file, appendFileName = true) {
   const fullFile = require.resolve(file)
   router.get(fullPath, (req, res) => res.sendFile(fullFile))
 }
+
+/**
+ * Returns the number of bytes that the user owning the requested POD
+ * may store or Infinity if no limit
+ */
+
+async function getQuota (root, serverUri) {
+  const filename = path.join(root, 'settings/serverSide.ttl')
+  var quota = Infinity
+  var prefs
+  try {
+    prefs = await _asyncReadfile(filename)
+  } catch (error) {
+    debug('Setting no quota. While reading serverSide.ttl, got ' + error)
+    return Infinity
+  }
+  var graph = $rdf.graph()
+  const storageUri = serverUri + '/'
+  try {
+    $rdf.parse(prefs, graph, storageUri, 'text/turtle')
+  } catch (error) {
+    throw new Error('Failed to parse serverSide.ttl, got ' + error)
+  }
+  const lit = graph.each($rdf.sym(storageUri), ns.solid('storageQuota'), undefined)[0]
+  if (lit) {
+    quota = lit.value
+    const unitUri = 'http://www.w3.invalid/ns#'
+    // The following should be encoded in the ontology
+    switch (lit.datatype.value) {
+      case unitUri + 'kilobyte':
+        quota *= 1000
+        break
+      case unitUri + 'megabyte':
+        quota *= 1000000
+        break
+      case unitUri + 'gigabyte':
+        quota *= 1000000000
+        break
+    }
+  }
+  return quota
+}
+
+/**
+ * Returns true of the user has already exceeded their quota, i.e. it
+ * will check if new requests should be rejected, which means they
+ * could PUT a large file and get away with it.
+ */
+
+async function overQuota (root, serverUri) {
+  let quota = await getQuota(root, serverUri)
+  if (quota === Infinity) {
+    return false
+  }
+  // TODO: cache this value?
+  var size = await actualSize(root)
+  console.log(size, quota)
```

Comment:
TODO: clarify

Reply:
Since `actualSize` is a heavy operation on large PODs (which the limit makes them not be for now), and it currently happens on every write, it is a DoS opportunity. Caching the value for a bit could alleviate that, but I don't think it is necessary yet. We need to do much better in v.next.